src/conn.rs

```diff
@@ -242,6 +257,22 @@ impl<'poll> Conn<'poll> {
 }
 
 impl<'poll> Conn<'poll> {
+    fn plain_sasl_authenticate(&mut self) {
+        if let (Some(stream), Some(auth)) =
+                (self.status.get_stream_mut(), self.auth.as_ref()) {
+            wire::authenticate(stream, "PLAIN").unwrap();
+            let msg =  format!("{}\x00{}\x00{}",
+                               auth.sasl_username, auth.sasl_username, auth.sasl_password);
+            wire::authenticate(stream, &base64::encode(&msg)).unwrap();
```

Comment:
Because these multiple messages should be sent together I think it makes sense to do this in `wire::authenticate`. Could you you send both `AUTHTENTICATE PLAIN` and `AUTHENTICATE <password>` in `wire::authenticate`?

Reply:
To support mozilla irc, this function is split into 2 pieces.